helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -428,13 +466,37 @@ public synchronized void setIdealStates(List<IdealState> idealStates) {
   }
 
   /**
-   * Returns the LiveInstances for each of the instances that are curretnly up and running
-   * @return
+   * Returns the LiveInstances for each of the instances that are currently up and running,
+   * excluding the instances that are considered offline during maintenance mode. Instances
+   * are timed-out if they have been offline for a while before going live during maintenance mode.
    */
   public Map<String, LiveInstance> getLiveInstances() {
+    return getLiveInstances(true);
+  }
+
+  /**
+   * Returns the LiveInstances for each of the instances that are currently up and running
+   */
+  public Map<String, LiveInstance> getAllLiveInstances() {
```

Comment:
These two methods (getLiveInstances() and getAllLiveInstances() are very confusing to each other.  We would need a better naming here.  We have four type of instances:  All Instances, Live Instances, enabled/disabled Instances, timeout-ed instances.  We have a method called getEnabledLiveInstances(), maybe we can name this new method getNonTimeoutLiveInstances()?

Reply:
I'm just going to remove this function and the boolean flag, since during maintenance mode, we always want to return the non-timed-out instances.